app/uk/gov/hmrc/gform/models/PageShader.scala

```diff
@@ -52,12 +52,19 @@ class PageShader(
     val section = dynamicSections(sectionNumber.value)
     for {
       snippets <- Future.sequence(section.fields.map(f => htmlFor(f, 0)))
-      javascript = fieldJavascript(dynamicSections.flatMap(_.atomicFields(repeatService)))
+      javascript = createJavascript(dynamicSections.flatMap(_.fields), dynamicSections.flatMap(_.atomicFields(repeatService)))
       hiddenTemplateFields = dynamicSections.filterNot(_ == section).flatMap(_.atomicFields(repeatService))
       hiddenSnippets = Fields.toFormField(fieldData, hiddenTemplateFields, repeatService).map(formField => uk.gov.hmrc.gform.views.html.hidden_field(formField))
     } yield PageForRender(formId, sectionNumber, section.title, section.description, hiddenSnippets, snippets, javascript, envelopeId)
   }
 
+  private def createJavascript(fieldList: List[FieldValue], atomicFields: List[FieldValue]) = {
+    val groups: List[(FieldId, Group)] = fieldList.collect{
+      case FieldValue(fieldId, group @Group(_, _, _, _, _, _), _, _, _, _, _, _, _, _) => (fieldId, group)
```

Comment:
I need to only match on group components not everything can pass through this validation.